ingestion/src/metadata/ingestion/source/messaging/common_broker_source.py

```diff
@@ -101,19 +104,47 @@ def yield_topic(
             self.add_properties_to_topic_from_resource(topic, topic_config_resource)
 
             if topic_schema is not None:
-                topic.schemaText = topic_schema.schema_str
                 if topic_schema.schema_type.lower() == SchemaType.Avro.value.lower():
-                    topic.schemaType = SchemaType.Avro.name
```

Comment:
These different logics based on the SchemaType could be a dispatch to keep implementations separated and here just have a single call

Reply:
I believe you could use this https://github.com/open-metadata/OpenMetadata/blob/main/ingestion/src/metadata/utils/dispatch.py#L26 if the `SchemaType` is an Enum

Reply:
Yes